usr/src/test/zfs-tests/cmd/dos_ro/dos_ro.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright 2017 Nexenta Systems, Inc.  All rights reserved.
+ */
+
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#include <attr.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <libnvpair.h>
+
+extern const char *__progname;
+
+int vflag = 0;
+
+static int
+dosattr_set_ro(int fildes, const char *fname)
+{
+	nvlist_t	*nvl = NULL;
+	int		err;
+
+	err = nvlist_alloc(&nvl, NV_UNIQUE_NAME, 0);
+	if (err != 0)
+		return (err);
+
+	(void) nvlist_add_boolean_value(nvl, A_READONLY, 1);
+
+	if (fname == NULL) {
+		err = fsetattr(fildes, XATTR_VIEW_READWRITE, nvl);
+	} else {
+		err = setattrat(fildes, XATTR_VIEW_READWRITE, fname, nvl);
+	}
+	if (err < 0) {
```

Comment:
Let's just leave it.  It might be useful to add a test case using that code path in the future.
(and I'm not too bothered by one line of dead code in a test program:)
Thanks!